Tell user, where the Golden Master was created

Currently, we produce the following output:
```
java.lang.AssertionError: 'com.mycompany.MyFirstTest':
No recheck file found. First time test was run? Created recheck file now, don't forget to commit...
        at de.retest.recheck.RecheckImpl.capTest(RecheckImpl.java:137)
        at com.mycompany.MyFirstTest.google(MyFirstTest.java:30)
        at ... some more stack trace ...
```
This should indicate _where_ the recheck files where created (i.e. path to `C:/...myproject/src/test/resources`).

--- a/src/main/java/de/retest/recheck/NoRecheckFileActionReplayResult.java
+++ b/src/main/java/de/retest/recheck/NoRecheckFileActionReplayResult.java
@@ -22,7 +22,7 @@ public class NoRecheckFileActionReplayResult extends ActionReplayResult {
 
 	private static final String MSG_SHORT = "No recheck file found.";
 	public static final String MSG_LONG =
-			MSG_SHORT + " First time test was run? Created recheck file now, don't forget to commit...";
+			MSG_SHORT + " First time test was run? Created new recheck file(s), so don't forget to commit...";
 
 	private static final long serialVersionUID = 1L;
 

--- a/src/main/java/de/retest/recheck/NoRecheckFileActionReplayResult.java
+++ b/src/main/java/de/retest/recheck/NoRecheckFileActionReplayResult.java
@@ -26,9 +26,10 @@ public class NoRecheckFileActionReplayResult extends ActionReplayResult {
 
 	private static final long serialVersionUID = 1L;
 
-	public NoRecheckFileActionReplayResult( final String currentStep, final SutState actual ) {
-		super( ActionReplayData.withoutTarget( currentStep, null ), WindowRetriever.empty(), ErrorHolder.empty(),
-				DifferenceRetriever.of( toStateDifference( actual ) ), 0L, null );
+	public NoRecheckFileActionReplayResult( final String currentStep, final SutState actual,
+			final String stateFilePath ) {
+		super( ActionReplayData.withoutTarget( currentStep, stateFilePath ), WindowRetriever.empty(),
+				ErrorHolder.empty(), DifferenceRetriever.of( toStateDifference( actual ) ), 0L, null );
 	}
 
 	private static StateDifference toStateDifference( final SutState actual ) {

--- a/src/main/java/de/retest/recheck/RecheckImpl.java
+++ b/src/main/java/de/retest/recheck/RecheckImpl.java
@@ -4,6 +4,7 @@ import java.io.File;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -134,8 +135,9 @@ public class RecheckImpl implements Recheck, SutStateLoader {
 		logger.info( "Found {} not ignored differences in test {}.", uniqueDifferences.size(),
 				finishedTestResult.getName() );
 		if ( !uniqueDifferences.isEmpty() ) {
-			final String message = finishedTestResult.hasNoRecheckFiles() ? getNoRecheckFilesErrorMessage()
-					: getDifferencesErrorMessage( finishedTestResult );
+			final String message =
+					finishedTestResult.hasNoRecheckFiles() ? getNoRecheckFilesErrorMessage( finishedTestResult )
+							: getDifferencesErrorMessage( finishedTestResult );
 			throw new AssertionError( message );
 		}
 	}
@@ -181,8 +183,11 @@ public class RecheckImpl implements Recheck, SutStateLoader {
 		}
 	}
 
-	private String getNoRecheckFilesErrorMessage() {
-		return "'" + suiteName + "':\n" + NoRecheckFileActionReplayResult.MSG_LONG;
+	private String getNoRecheckFilesErrorMessage( final TestReplayResult finishedTestResult ) {
+		final String stateFilePaths = finishedTestResult.getActionReplayResults().stream() //
+				.map( ActionReplayResult::getStateFilePath ) //
+				.collect( Collectors.joining( "\n" ) );
+		return "'" + suiteName + "':\n" + NoRecheckFileActionReplayResult.MSG_LONG + "\n" + stateFilePaths;
 	}
 
 	private String getDifferencesErrorMessage( final TestReplayResult finishedTestResult ) {

--- a/src/main/java/de/retest/recheck/RecheckImpl.java
+++ b/src/main/java/de/retest/recheck/RecheckImpl.java
@@ -102,7 +102,7 @@ public class RecheckImpl implements Recheck, SutStateLoader {
 		final SutState expected = loadExpected( file );
 		if ( expected == null ) {
 			createNew( file, actual );
-			return new NoRecheckFileActionReplayResult( currentStep, actual );
+			return new NoRecheckFileActionReplayResult( currentStep, actual, file.getPath() );
 		}
 		final RecheckDifferenceFinder finder =
 				new RecheckDifferenceFinder( adapter.getDefaultValueFinder(), currentStep, file.getPath() );

--- a/src/test/java/de/retest/recheck/RecheckImplTest.java
+++ b/src/test/java/de/retest/recheck/RecheckImplTest.java
@@ -126,7 +126,7 @@ class RecheckImplTest {
 		private File resolveRoot( final String[] baseNames, final String extension ) {
 			final int last = baseNames.length - 1;
 			final List<String> list = new ArrayList<>( Arrays.asList( baseNames ) );
-			list.set( last, baseNames[last] + "." + extension );
+			list.set( last, baseNames[last] + extension );
 
 			Path path = root;
 			for ( final String sub : list ) {

--- a/src/test/java/de/retest/recheck/RecheckImplTest.java
+++ b/src/test/java/de/retest/recheck/RecheckImplTest.java
@@ -68,13 +68,15 @@ class RecheckImplTest {
 		when( adapter.canCheck( any() ) ).thenReturn( true );
 		when( adapter.convert( any() ) ).thenReturn( Collections.singleton( rootElement ) );
 
-		cut.startTest( "Some test" );
-		cut.check( "Some object to verify", adapter, "Some step" );
+		cut.startTest( "some-test" );
+		cut.check( "to-verify", adapter, "some-step" );
 
+		final String goldenMasterName = "de.retest.recheck.RecheckImplTest/some-test.some-step.recheck";
 		assertThatThrownBy( cut::capTest ) //
 				.isExactlyInstanceOf( AssertionError.class ) //
-				.hasMessageStartingWith( "'de.retest.recheck.RecheckImplTest':\n"
-						+ "No recheck file found. First time test was run? Created recheck file now, don't forget to commit..." );
+				.hasMessageStartingWith(
+						"'" + getClass().getName() + "':\n" + NoRecheckFileActionReplayResult.MSG_LONG ) //
+				.hasMessageEndingWith( goldenMasterName );
 
 	}
 

--- a/src/test/java/de/retest/recheck/printer/ActionReplayResultPrinterTest.java
+++ b/src/test/java/de/retest/recheck/printer/ActionReplayResultPrinterTest.java
@@ -97,11 +97,13 @@ class ActionReplayResultPrinterTest {
 		final SutState state = mock( SutState.class );
 		when( state.getRootElements() ).thenReturn( Collections.singletonList( element ) );
 
-		final NoRecheckFileActionReplayResult result = new NoRecheckFileActionReplayResult( "foo", state );
+		final String stateFilePath = "some/state/file/path";
+
+		final NoRecheckFileActionReplayResult result =
+				new NoRecheckFileActionReplayResult( "foo", state, stateFilePath );
 
 		final String string = cut.toString( result );
 
-		assertThat( string ).contains(
-				"No recheck file found. First time test was run? Created recheck file now, don't forget to commit..." );
+		assertThat( string ).contains( NoRecheckFileActionReplayResult.MSG_LONG );
 	}
 }

